woodwork/statistics_utils/_get_dependence_dict.py

```diff
@@ -0,0 +1,278 @@
+import warnings
+from collections import defaultdict
+from timeit import default_timer as timer
+
+import numpy as np
+import pandas as pd
+from sklearn.metrics.cluster import adjusted_mutual_info_score
+
+from ._bin_numeric_cols_into_categories import _bin_numeric_cols_into_categories
+
+from woodwork.accessor_utils import _is_dask_dataframe, _is_koalas_dataframe
+from woodwork.exceptions import SparseDataWarning
+from woodwork.logical_types import IntegerNullable
+from woodwork.utils import CallbackCaller, get_valid_mi_types, get_valid_pearson_types
+
+
+def _get_dependence_dict(
+    dataframe,
+    measures,
+    num_bins=10,
+    nrows=None,
+    include_index=False,
+    callback=None,
+    extra_stats=False,
+    min_shared=25,
+    random_seed=0,
+):
+    """Calculates dependence measures between all pairs of columns in the DataFrame that
+    support measuring dependence. Supports boolean, categorical, datetime, and numeric data.
+    Call woodwork.utils.get_valid_dependence_types for a complete list of supported Logical Types.
```

Comment:
Should probably change to

Call woodwork.utils.get_valid_mi_types  and woodwork.utils.get_valid_pearson_types for complete lists of supported Logical Types.

(and update wording similarly in table_accessor.py)